src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py

```diff
@@ -903,6 +907,128 @@ def __init__(self, dataset_params=None, cache_labels: bool = False, cache_images
         self.classes = self.trainset.classes
 
 
+class PascalVOCUnifiedDetectionDataSetInterfaceV2(DatasetInterface):
+
+    def __init__(self, dataset_params=None):
+        if dataset_params is None:
+            dataset_params = dict()
+        super().__init__(dataset_params=dataset_params)
+
+        self.data_dir = core_utils.get_param(self.dataset_params, 'data_dir', "~/data/pascal_unified_coco_format/")
+
+        cache_dir = core_utils.get_param(self.dataset_params, "cache_dir", self.data_dir)
+        cache_train_images = core_utils.get_param(self.dataset_params, 'cache_train_images', default_val=True)
+        cache_val_images = core_utils.get_param(self.dataset_params, 'cache_val_images', default_val=True)
+
+        class_inclusion_list = core_utils.get_param(self.dataset_params, "class_inclusion_list")
+
+        train_input_dim = (self.dataset_params.train_image_size, self.dataset_params.train_image_size)
+        val_input_dim = (self.dataset_params.val_image_size, self.dataset_params.val_image_size)
+
+        train_transforms = core_utils.get_param(self.dataset_params, 'train_transforms',
+                                                default_val=[DetectionMosaic(input_dim=train_input_dim, prob=1),
```

Comment:
There are 2 ways to setup the transforms:
 - either we pre-define a pipeline of functions and we accept the parameters of these functions as dataset_params (the way cocoV2 is at the moment)
 - or we expect the user to build his own pipeline in the yaml with a list of _target_ for every transform he wants to use.

I understood that we want the second option and this is what this code is coded for. I just left the default values that were associated to pascal but I agree that this is not a good idea.
So is it good if I just remove "default_val" (i.e. we go with the second approach and expect the user to call the transform he wants) or should I go to the first approach ?

Reply:
Your terminology is wrong. Its not that we expect the user to do so with list of target, but we are aiming to define everything by the YAML files (which can be easily written now that we have more and more factories). As for user defined transforms- this is something we will also solve so that it'd be easy to configure from the YAML file as well in the future.

In short- we should define the transforms also in the yaml file, so yes it would be nice to have them configured and not hard coded, though its not mandatory for this PR in my opinion.